src/ui/page_selector.rs

```diff
@@ -83,6 +89,9 @@ impl PageSelector {
                         &mut page_button.button_state,
                         Text::new(page_button.label_text.clone()).size(16),
                     )
+                    .style(style::Button::Disabled {
+                        disabled: vec![page_button.page_type] == *disabled_pages,
```

Comment:
Why are you comparing the two vecs? Shouldn't you be checking to see if it's in the vec? Hm... Perhaps an enum with a match statement will be more idiomatic... We'll have to think of that on a call.

Reply:
Yeah, let's do that